packages/core/src/html/linkProcessor.ts

```diff
@@ -23,34 +28,36 @@ const defaultTagLinkMap = {
   script: 'src',
 };
 
-function hasTagLink(node) {
-  return node.name in defaultTagLinkMap || node.name in pluginTagConfig;
+export function hasTagLink(node: DomElement) {
+  return node.name && (node.name in defaultTagLinkMap || node.name in pluginTagConfig);
 }
 
-function getDefaultTagsResourcePath(node) {
+export function getDefaultTagsResourcePath(node: DomElement): string {
+  if (!node.name || !node.attribs) return '';
   const linkAttribName = defaultTagLinkMap[node.name];
-  const resourcePath = node.attribs && node.attribs[linkAttribName];
+  const resourcePath = node.attribs[linkAttribName];
   return resourcePath;
 }
 
-function getResourcePathFromRoot(rootPath, fullResourcePath) {
+function _getResourcePathFromRoot(rootPath: string, fullResourcePath: string) {
   return fsUtil.ensurePosix(path.relative(rootPath, fullResourcePath));
 }
 
 /**
  * @param {string} resourcePath parsed from the node's relevant attribute
  * @returns {boolean} whether the resourcePath is a valid intra-site link
  */
-function isIntraLink(resourcePath) {
+export function isIntraLink(resourcePath: string | undefined): boolean {
   const MAILTO_OR_TEL_REGEX = /^(?:mailto:|tel:)/i;
-  return resourcePath
+  return !!resourcePath
```

Comment:
Just to check, what does `!!resourcePath` mean?

Reply:
It converts `resourcePath` to a boolean (like `!(!resourcePath)`) so that our return types are standardised :)